docs/DeveloperGuide.md

```diff
@@ -1,38 +1,662 @@
+2. time: Represents the time of the appointment.
+3. description: Stores a description of the appointment.
+
+**Methods:**
+1. Appointment(String stringDate, String stringTime, String description): The constructor of Appointment which takes date, time, and description.
+2. getDate(): Retrieves the date of the appointment.
+3. getTime(): Retrieves the time of the appointment.
+4. getDescription(): Retrieves the description of the appointment.
+5. toString(): Returns a string representation of the Appointment object, including the date, time, and description.
+
+###### [Back to table of contents](#table-of-contents)
+
+### Utility Package
+
+The `Utility` package includes classes and methods that handle exceptions, user input parsing, user input validation, and the various filter strings using enumerations.
+
+It consists of `CustomExceptions`, `Filters`, `Parser` and `Validation` classes.
+
+{Include `Utility` class diagram}
+
+###### [Back to table of contents](#table-of-contents)
+
+#### Parser
+
+{Include `Parser` description}
+
+###### [Back to table of contents](#table-of-contents)
+
+#### Validation
+
+{Include `Validation` description}
+
+###### [Back to table of contents](#table-of-contents)
+
+#### Custom Exceptions
+
+{Include `CustomException` description}
+
+###### [Back to table of contents](#table-of-contents)
+
+#### Filters
+
+{Include `Filters` description}
+
+###### [Back to table of contents](#table-of-contents)
+
+### Storage Package
+
+`Storage` contains `DataFile` and `LogFile`. This component handles all logging of commands used and writing of data stored within PulsePilot to an external data file. The reading of the data file is also done here.
+
+{Include `Storage` class diagram}
+
+The Storage component consists of LogFile, DataFile, and DataType.
+
+###### [Back to table of contents](#table-of-contents)
+
+#### Log File
+
+###### [Back to table of contents](#table-of-contents)
+
+#### Data File
+
+###### [Back to table of contents](#table-of-contents)
+
+### PulsePilot Package
+
+###### [Back to table of contents](#table-of-contents)
+
+### Constants Package
+
+###### [Back to table of contents](#table-of-contents)
+
+## Commands and Implementation
+
+* [Workout](#workout)
+    * [Add Run](#add-run)
+    * [Add Gym](#add-gym)
+* [Health](#health)
+    * [Add Period](#add-period)
+    * [Add BMI](#add-bmi)
+    * [Add Appointment](#add-appointment)
+    * [Make Period Prediction](#make-period-prediction)
+* [View History](#view-history)
+* [View Latest](#view-latest)
+* [Delete Item](#delete-item)
+* [Storage of Data](#storage-of-data)
+
+### Workout
+
+#### Add Run
+
+<code style="color: #D85D43;">
+WORKOUT /e:run /d:[distance] /t:[time] /date:[date]
+</code>
+
+- `[distance]` is a 2 **decimal place positive number** representing the number of kilometers covered.
+- `[time]` is in `HH:MM:SS` or `MM:SS` format with positive integers, representing the amount of time taken for the run.
```

Comment:
sorry if i missed it but maybe clarify whats the type

![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/61624545/cdbbda49-1095-472d-828d-30a0bcbc38ed)